internal/bminventory/inventory.go

```diff
@@ -5551,3 +5553,39 @@ func (b *bareMetalInventory) V2PostStepReply(ctx context.Context, v2Params insta
 func (b *bareMetalInventory) V2GetHost(ctx context.Context, params installer.V2GetHostParams) middleware.Responder {
 	return installer.NewV2GetHostNotImplemented()
 }
+
+func (b *bareMetalInventory) BindHost(ctx context.Context, params installer.BindHostParams) middleware.Responder {
+	log := logutil.FromContext(ctx, b.log)
+	log.Infof("Bind host %s to cluster %s", params.HostID, params.BindHostParams.ClusterID)
+	host, err := common.GetHostFromDB(b.db, params.InfraEnvID.String(), params.HostID.String())
+	if err != nil {
+		log.WithError(err).Errorf("failed to find host <%s> in infraEnv <%s>",
+			params.HostID, params.InfraEnvID)
+		return common.NewApiError(http.StatusNotFound, err)
+	}
+	if host.ClusterID != nil {
+		return common.NewApiError(http.StatusConflict, errors.Errorf("Host %s is already bound to cluster %s", params.HostID, *host.ClusterID))
+	}
+	cluster, err := common.GetClusterFromDB(b.db, *params.BindHostParams.ClusterID, common.SkipEagerLoading)
+	if err != nil {
+		return common.NewApiError(http.StatusBadRequest, errors.Errorf("Failed to find cluster %s", params.BindHostParams.ClusterID))
+	}
+	if err = b.clusterApi.AcceptRegistration(cluster); err != nil {
+		log.WithError(err).Errorf("failed to bind host <%s> to cluster %s due to: %s",
+			params.HostID.String(), *params.BindHostParams.ClusterID, err.Error())
+		return common.NewApiError(http.StatusConflict, err)
+	}
+
+	if err = b.hostApi.BindHost(ctx, &host.Host, *params.BindHostParams.ClusterID, b.db); err != nil {
+		log.WithError(err).Errorf("Failed to bind host <%s> to cluster <%s>",
+			params.HostID, *params.BindHostParams.ClusterID)
+		return common.NewApiError(http.StatusInternalServerError, err)
+	}
+
+	host, err = common.GetHostFromDB(b.db, params.InfraEnvID.String(), params.HostID.String())
```

Comment:
Why is this required?
You got the host from DB, the `b.hostApi.BindHost` updated it, I think you can just return it.

Reply:
I'm not sure how the state machine works, are you sure that it updated the object I sent to it?
@filanov can you confirm that?

Reply:
state machine updates the host in DB, and then updates the pointer to the host in stateHost, and not the host pointed by stateHost, so we need to read from DB if we want to get the right cluster id in the returned host

Reply:
/lgtm